src/library/searchqueryparser.cpp

```diff
@@ -291,6 +291,18 @@ std::unique_ptr<OrNode> SearchQueryParser::parseOrNode(const QString& query) con
 #endif
     for (const QString& rawAndNode : rawAndNodes) {
         if (!rawAndNode.isEmpty()) {
+            // Try both apostrophes and right single quotation marks.
+            if (rawAndNode.contains(QChar('\''))) {
+                QString quotationMarkAndNode = rawAndNode;
+                quotationMarkAndNode.replace(QChar('\''), QChar(0x2019));
+                pQuery->addNode(parseAndNode(quotationMarkAndNode));
+            }
+            if (rawAndNode.contains(QChar(0x2019))) {
+                QString apostropheAndNode = rawAndNode;
+                apostropheAndNode.replace(QChar(0x2019), QChar('\''));
+                pQuery->addNode(parseAndNode(apostropheAndNode));
+            }
```

Comment:
We should be a bit careful about performing substitutions on the pre-tokenized input. With these characters it's likely to be fine, but e.g. `"` is used to quote strings.

I also feel like this could be generalized. Instead of hardcoding substitutions, perhaps there is a way to perform a fully symbol-insensitive search (unless e.g. the query is quoted)?

IIRC there were also aalready some earlier discussions on adding 'actual' fuzzy search, which would make these tricks largely irrelevant since most parts of the string would still match:

- #7470

Reply:
I was conscious of that, but it's much simpler to do it before than trying to copy and modify the already-created nodes. At least currently it wouldn't be an issue with `'`.

It could be done by modifying the SQL queries to `REPLACE()` certain symbols or regex could be used in the SQL to strip out symbols in general, but I was concerned by the performance implications and honestly couldn't think of other common symbols, although I'm sure I'm missing some. I guess words may be hyphenated or not.

Fuzzy search would be an alternative, but my understanding is that it will require prepending a tilde (e.g., `~that's`) to match. I doubt most people will know there are multiple types of apostrophes to know they may need to use fuzzy matching to find something. Plus, it's just more broad, and you don't always want "close" words just because you have apostrophes.

I will write some tests though. Another thought is that I think using the [ICU SQLite extension](https://www.sqlite.org/src/artifact?ci=trunk&filename=ext/icu/README.txt) for normalization would be the best way to treat equivalent symbols the same, and I think that would work without editing the search code since I think it just replaces `LIKE` under the hood. I'm just less familiar with that. I'll look into that as a more general alternative.

Thanks for your feedback!